table/column.go

```diff
@@ -391,7 +391,7 @@ func getColDefaultExprValue(ctx sessionctx.Context, col *model.ColumnInfo, defau
 		return types.Datum{}, err
 	}
 	// Check the evaluated data type by cast.
-	value, err := CastValue(ctx, types.NewDatum(d), col)
+	value, err := CastValue(ctx, d, col)
```

Comment:
Any reason for this change?

Reply:
d is already the datum of kindInt64, no need to compact it.